mlearning/cmsis/libcmsisdsp/Makefile

```diff
@@ -0,0 +1,556 @@
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_cfft_radix4_init_f32.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_cfft_radix4_q31.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/TransformFunctions.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_cfft_radix2_init_q31.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_cfft_radix2_init_f16.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_rfft_fast_init_f32.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_bitreversal_f16.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_cfft_init_q15.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_rfft_fast_f16.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_cfft_radix8_f16.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_rfft_fast_init_f16.c
+CSRCS += $(CMSIS5_DSP)/TransformFunctions/arm_rfft_q15.c
+CSRCS += $(CMSIS5_DSP)/BayesFunctions/BayesFunctionsF16.c
+CSRCS += $(CMSIS5_DSP)/BayesFunctions/arm_gaussian_naive_bayes_predict_f32.c
+CSRCS += $(CMSIS5_DSP)/BayesFunctions/arm_gaussian_naive_bayes_predict_f16.c
+CSRCS += $(CMSIS5_DSP)/BayesFunctions/BayesFunctions.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mag_squared_f16.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mag_squared_q15.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mag_squared_q31.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mult_cmplx_f32.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_conj_q15.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mult_real_f32.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mag_q15.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_dot_prod_f32.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mult_cmplx_f16.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mult_cmplx_q31.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mag_q31.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mult_real_q15.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_dot_prod_q15.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_conj_f16.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_conj_f32.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mag_squared_f32.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mult_cmplx_q15.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_dot_prod_q31.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/ComplexMathFunctions.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mag_f16.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/ComplexMathFunctionsF16.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mult_real_f16.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_dot_prod_f16.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_conj_q31.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mult_real_q31.c
+CSRCS += $(CMSIS5_DSP)/ComplexMathFunctions/arm_cmplx_mag_f32.c
+
+CFLAGS += -fno-strict-aliasing -ffunction-sections -fdata-sections
+CFLAGS += -ffast-math -Wno-undef -Wno-shadow -Wno-undef
```

Comment:
remove the last dup -Wno-undef, and should we define the macro to the known value and then remove -Wno-undef?